### PR TITLE
Add user guide to new Fast Pair enable API

### PR DESCRIPTION
**Pull request by alstrzebonski**
Aligns the Fast Pair documentation after new Fast Pair enable API was introduced via this PR: https://github.com/nrfconnect/sdk-nrf/pull/13315.

### CODE REVIEW

**Review by mkapala-nordic — APPROVED** [review]

File: doc/nrf/external_comp/bt_fast_pair.rst [hunk 1]
```diff
@@ -304,7 +311,7 @@ Both functions are defined as weak no-op functions.
 Ensure that your reset action implementation executes correctly in the following execution contexts:
 
 * In the :c:func:`bt_fast_pair_factory_reset` function context: the factory reset action is triggered by calling the :c:func:`bt_fast_pair_factory_reset` function.
-* In the :c:func:`settings_load` function context during the commit phase (after data is loaded from the non-volatile memory): the factory reset action using the :c:func:`bt_fast_pair_factory_reset` function was interrupted and the factory reset is retried on the system bootup.
+* In the :c:func:`bt_fast_pair_enable` function context: the factory reset action using the :c:func:`bt_fast_pair_factory_reset` function was interrupted and the factory reset is retried when enabling the Fast Pair subsystem.
```
Comment (divipillai) [inline doc/nrf/external_comp/bt_fast_pair.rst:314]:
```suggestion
* In the :c:func:`bt_fast_pair_enable` function context - The factory reset action using the :c:func:`bt_fast_pair_factory_reset` function was interrupted, and the factory reset is retried when enabling the Fast Pair subsystem.
```

File: doc/nrf/external_comp/bt_fast_pair.rst [hunk 2]
```diff
@@ -290,6 +294,9 @@ Managing factory resets
 
 The Fast Pair GATT service uses a non-volatile memory to store the Fast Pair user data such as Account Keys and the Personalized Name.
 This data can be cleared by calling the :c:func:`bt_fast_pair_factory_reset` function.
+Calling the :c:func:`bt_fast_pair_factory_reset` function doesn't affect the Fast Pair subsystem readiness.
```
Comment (divipillai) [inline doc/nrf/external_comp/bt_fast_pair.rst:297]:
```suggestion
Calling the :c:func:`bt_fast_pair_factory_reset` function does not affect the Fast Pair subsystem's readiness.
```

File: doc/nrf/external_comp/bt_fast_pair.rst [hunk 3]
```diff
@@ -290,6 +294,9 @@ Managing factory resets
...
+If the subsystem is enabled with the :c:func:`bt_fast_pair_enable` function, it will stay enabled after calling the :c:func:`bt_fast_pair_factory_reset` function.
```
Comment (divipillai) [inline doc/nrf/external_comp/bt_fast_pair.rst:298]:
```suggestion
If the subsystem is enabled with the :c:func:`bt_fast_pair_enable` function, it stays enabled after calling the :c:func:`bt_fast_pair_factory_reset` function.
```

File: doc/nrf/external_comp/bt_fast_pair.rst [hunk 4]
```diff
@@ -304,7 +311,7 @@ Both functions are defined as weak no-op functions.
 Ensure that your reset action implementation executes correctly in the following execution contexts:
 
 * In the :c:func:`bt_fast_pair_factory_reset` function context: the factory reset action is triggered by calling the :c:func:`bt_fast_pair_factory_reset` function.
```
Comment (divipillai) [inline doc/nrf/external_comp/bt_fast_pair.rst:313]:
```suggestion
* In the :c:func:`bt_fast_pair_factory_reset` function context - The factory reset action is triggered by calling the :c:func:`bt_fast_pair_factory_reset` function.
```

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 5]
```diff
@@ -742,6 +742,11 @@ Bluetooth libraries and services
 
 * :ref:`bt_fast_pair_readme` library:
 
+  * Added:
+
+    * The :c:func:`bt_fast_pair_enable`, :c:func:`bt_fast_pair_disable`, :c:func:`bt_fast_pair_is_ready` functions to handle the Fast Pair subsystem readiness.
+      In order to use the Fast Pair service, the Fast Pair subsystem must be enabled with the :c:func:`bt_fast_pair_enable` function.
```
Comment (divipillai) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:748]:
```suggestion
  * Added the :c:func:`bt_fast_pair_enable`, :c:func:`bt_fast_pair_disable`, :c:func:`bt_fast_pair_is_ready` functions to handle the Fast Pair subsystem readiness.
    In order to use the Fast Pair service, the Fast Pair subsystem must be enabled with the :c:func:`bt_fast_pair_enable` function.
```

**Comment by alstrzebonski** [comment]
Added small API description alignment.

**Review by divipillai — APPROVED** [review]

File: doc/nrf/external_comp/bt_fast_pair.rst [hunk 6]
```diff
@@ -166,7 +166,11 @@ The standard Fast Pair API (without extensions) currently supports the :c:func:`
 
 The standard Fast Pair (without extensions) does not require registration of any callback type, meaning all callbacks are optional.
 
-Apart from the callback registration, no additional operations are needed to integrate the standard Fast Pair implementation.
+After the callback registration, the Fast Pair subsystem must be enabled with the :c:func:`bt_fast_pair_enable` function.
+The Fast Pair subsystem readiness can be checked with the :c:func:`bt_fast_pair_is_ready` function.
+The Fast Pair subsystem can be disabled with the :c:func:`bt_fast_pair_disable` function.
+
+Apart from the callback registration and enabling the Fast Pair subsystem, no additional operations are needed to integrate the standard Fast Pair implementation.
```
Comment (MarekPieta) [inline doc/nrf/external_comp/bt_fast_pair.rst:173]: Consider mentioning that user needs to enable Bluetooth and load settings before enabling Fast Pair.

Comment (MarekPieta) [reply doc/nrf/external_comp/bt_fast_pair.rst:173]: `to integrate the core Fast Pair implementation`?

File: applications/nrf_desktop/doc/fast_pair_app.rst [hunk 7]
```diff
@@ -64,5 +64,5 @@ This allows the module to update the Fast Pair advertising payload just before t
 The module is a subscriber for :c:struct:`ble_dongle_peer_event`.
 This allows the module to remove the Fast Pair advertising payload when the application identity of the dongle peer is used.
 
-The module registers the global application's Bluetooth authentication callbacks (:c:struct:`bt_conn_auth_cb`) on application start.
+The module registers the global application's Bluetooth authentication callbacks (:c:struct:`bt_conn_auth_cb`) and enables the :ref:`bt_fast_pair_readme` (:c:func:`bt_fast_pair_enable`) after Zephyr's :ref:`zephyr:settings_api` are loaded.
```
Comment (MarekPieta) [inline applications/nrf_desktop/doc/fast_pair_app.rst:67]: to be more precise and add a ref to initialization chain
```
after :ref:`caf_settings_loader` loads Zephyr's :ref:`zephyr:settings_api`.
```

**Review by MarekPieta — APPROVED** [review]

File: doc/nrf/external_comp/bt_fast_pair.rst [hunk 8]
```diff
@@ -166,7 +166,11 @@ The standard Fast Pair API (without extensions) currently supports the :c:func:`
 
 The standard Fast Pair (without extensions) does not require registration of any callback type, meaning all callbacks are optional.
 
-Apart from the callback registration, no additional operations are needed to integrate the standard Fast Pair implementation.
+After the callback registration, the Fast Pair subsystem must be enabled with the :c:func:`bt_fast_pair_enable` function.
+The Fast Pair subsystem readiness can be checked with the :c:func:`bt_fast_pair_is_ready` function.
+The Fast Pair subsystem can be disabled with the :c:func:`bt_fast_pair_disable` function.
```
Comment (kapi-no) [inline doc/nrf/external_comp/bt_fast_pair.rst:171]: Consider also mentioning the effects of having the subsystem in the disabled state:

- One sentence here: `In the Fast Pair subsystem disabled state, most Fast Pair APIs are not available`

- One sentence in the Interacting with GATT service section. For example: `In the Fast Pair subsystem disabled state, GATT operations on the Fast Pair service are rejected`.

File: doc/nrf/external_comp/bt_fast_pair.rst (hunk 6 above)
Comment (kapi-no) [reply doc/nrf/external_comp/bt_fast_pair.rst:173]: > `to integrate the core Fast Pair implementation`?

We have a few mentions (L165, L167) of the standard Fast Pair implementation in this RST file. Calling it here the core Fast Pair implementation would confuse the user. We agreed in previous code reviews to use the "standard" keyword to refer to the core part. I think we should keep it as it is

Comment (kapi-no) [reply doc/nrf/external_comp/bt_fast_pair.rst:173]: My comment was about keeping the consistency. The confusion here stems from the fact that we use two different words to refer to the same thing.

Is there a `core` use case? Both "core" and "standard" are made-up words by us and are not used in the original Fast Pair specification. When I say that we agreed to the previous wording, I have the following PR in mind:

https://github.com/nrfconnect/sdk-nrf/pull/13229

where there were no comments regarding this wording

I will leave it up to @alstrzebonski to select the approach for this one. However, if we decide to go with the "core" word, then we need to align it consistently in this guide.

Comment (alstrzebonski) [reply doc/nrf/external_comp/bt_fast_pair.rst:173]: I will leave it as is. I think it's not confusing and reader can easily understand what we mean.

**Comment by alstrzebonski** [comment]
Addressed review comments.

File: doc/nrf/external_comp/bt_fast_pair.rst [hunk 9]
```diff
@@ -154,19 +154,26 @@ Performing prerequisite operations
 
 You must enable the :kconfig:option:`CONFIG_BT_FAST_PAIR` Kconfig option to support the Google Fast Pair standard in your project.
 
+The Fast Pair subsystem requires enabling Bluetooth with the :c:func:`bt_enable` function and loading Zephyr's :ref:`zephyr:settings_api` with the :c:func:`settings_load` function.
```
Comment (kapi-no) [inline doc/nrf/external_comp/bt_fast_pair.rst:157]: Maybe we can put this section in the proximity of L171, as you don't need to perform these operations before calling register APIs. You just need to prepare these dependencies before you enable the Fast Pair subsystem:

```
After the callback registration, the Fast Pair subsystem must be enabled with the :c:func:`bt_fast_pair_enable` function.
Before the :c:func:`bt_fast_pair_enable` operation, you must enable Bluetooth with the :c:func:`bt_enable` function and load Zephyr's :ref:`zephyr:settings_api` with the :c:func:`settings_load` function.
```

**Comment by alstrzebonski** [comment]
Addressed review comment.

**Review by kapi-no — APPROVED** [review]

File: doc/nrf/external_comp/bt_fast_pair.rst [hunk 10]
```diff
@@ -157,16 +157,22 @@ You must enable the :kconfig:option:`CONFIG_BT_FAST_PAIR` Kconfig option to supp
...
+The standard Fast Pair API (without extensions) currently supports the :c:func:`bt_fast_pair_info_cb_register` function (optional) for registering application callbacks.
 
 The standard Fast Pair (without extensions) does not require registration of any callback type, meaning all callbacks are optional.
 
-Apart from the callback registration, no additional operations are needed to integrate the standard Fast Pair implementation.
+After the callback registration, the Fast Pair subsystem must be enabled with the :c:func:`bt_fast_pair_enable` function.
+Before performing the :c:func:`bt_fast_pair_enable` operation, you must enable Bluetooth with the :c:func:`bt_enable` function and load Zephyr's :ref:`zephyr:settings_api` with the :c:func:`settings_load` function.
+The Fast Pair subsystem readiness can be checked with the :c:func:`bt_fast_pair_is_ready` function.
+The Fast Pair subsystem can be disabled with the :c:func:`bt_fast_pair_disable` function.
+In the Fast Pair subsystem disabled state, most of the Fast Pair APIs are not available.
```
Comment (MarekPieta) [inline doc/nrf/external_comp/bt_fast_pair.rst:173]: Seems like the second part of @kapi-no's comment was overlooked. Consider adding it as follows:

`The Fast Pair GATT Service is statically defined, so it's still present in GATT database after Fast Pair is disabled. In the Fast Pair subsystem disabled state, GATT operations on the Fast Pair service are rejected.

File: doc/nrf/external_comp/bt_fast_pair.rst [hunk 11]
```diff
@@ -263,6 +269,8 @@ The service implements functionalities required by the `Fast Pair Procedure`_.
 The procedure is initiated by the Fast Pair Seeker after Bluetooth LE connection is established.
 No application interaction is required.
 
+In the Fast Pair subsystem disabled state, GATT operations on the Fast Pair service are rejected.
```
Comment (MarekPieta) [inline doc/nrf/external_comp/bt_fast_pair.rst:272]: The last small improvement (would be good to explain why it's implemented like that I think)

```
The Fast Pair GATT Service is statically defined, so it's still present in GATT database after Fast Pair is disabled. In the Fast Pair subsystem disabled state, GATT operations on the Fast Pair service are rejected.
```

**Review by MarekPieta — APPROVED** [review]

File: doc/nrf/external_comp/bt_fast_pair.rst (hunk 11 above)
Comment (alstrzebonski) [reply doc/nrf/external_comp/bt_fast_pair.rst:273]: Addressed.

File: doc/nrf/external_comp/bt_fast_pair.rst [hunk 12]
```diff
@@ -263,6 +269,9 @@ The service implements functionalities required by the `Fast Pair Procedure`_.
 The procedure is initiated by the Fast Pair Seeker after Bluetooth LE connection is established.
 No application interaction is required.
 
+The Fast Pair GATT Service is statically defined, so it's still present in the GATT database after the Fast Pair subsystem is disabled.
```
Comment (divipillai) [inline doc/nrf/external_comp/bt_fast_pair.rst:272]:
```suggestion
The Fast Pair GATT service is statically defined, so it is still present in the GATT database after the Fast Pair subsystem is disabled.
```

**Review by divipillai — APPROVED** [review]
nit

**Review by MarekPieta — APPROVED** [review]

**Pull request merged**